### PR TITLE
Improve template system and panic recovery

### PR DESCRIPTION
**Pull request by wxiaoguang**
Partially for #24457

Major changes:

1. The old `signedUserNameStringPointerKey` is quite hacky, use `ctx.Data[SignedUser]` instead
2. Move duplicate code from `Contexter` to `CommonTemplateContextData`
3. Remove incorrect copying&pasting code `ctx.Data["Err_Password"] = true` in API handlers
4. Use one unique `RenderPanicErrorPage` for panic error page rendering
5. Move `stripSlashesMiddleware` to be the first middleware
6. Install global panic recovery handler, it works for both `install` and `web`
7. Make `500.tmpl` only depend minimal template functions/variables, avoid triggering new panics

Screenshot:

<details>

image

</details>

### CODE REVIEW

File: templates/status/500.tmpl [hunk 1]
```diff
@@ -1,36 +1,61 @@
...
+		if (embedded) {
+			// move the 500 error page content to main view
+			const embeddedParent = embedded.parentNode;
+			let main = document.querySelector('.page-content');
+			main = main ?? document.querySelector('body');
+			main.prepend(document.createElement('hr'));
+			main.prepend(embedded);
+			embeddedParent.remove(); // remove the unrelated 500-page elements (eg: the duplicate nav bar)
+		}
+	</script>
```
Comment (silverwind) [inline templates/status/500.tmpl:59]: Does this have to be inline? I think not as `index.js` still loads on these pages, right?

Comment (wxiaoguang) [reply templates/status/500.tmpl:59]: No, there is no `<script>` inclusion in this page now.

And, this code is only for 500 page, quite hacky, I do not think it should be put into `index.js`

Comment (silverwind) [reply templates/status/500.tmpl:59]: Could create a webpack chunk `standalone/error.js`. Better than inline at least.

Comment (wxiaoguang) [reply templates/status/500.tmpl:59]: I do not see why it's necessary. 0edca5125d0a77e9ed8448a061e49a1024a911d3

Comment (silverwind) [reply templates/status/500.tmpl:59]: Maybe later

**Review by silverwind — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**